kolibri/plugins/coach/assets/src/state/actions/exam.js

```diff
@@ -101,111 +148,80 @@ function showExamsPage(store, classId) {
   );
 }
 
+// fetches topic, it's children subtopics, and children exercises
+function fetchContent(store, channelId, topicId) {
+  return new Promise((resolve, reject) => {
+    const channelPayload = { channel_id: channelId };
+    const topicPromise = ContentNodeResource.getModel(topicId, channelPayload).fetch();
+    const subtopicsPromise = ContentNodeResource.getCollection(
+      channelPayload, { parent: topicId, kind: ContentNodeKinds.TOPIC }).fetch();
+    const exercisesPromise = ContentNodeResource.getCollection(
+      channelPayload, { parent: topicId, kind: ContentNodeKinds.EXERCISE }).fetch();
+
+    ConditionalPromise.all([topicPromise, subtopicsPromise, exercisesPromise]).only(
+      CoreActions.samePageCheckGenerator(store),
+      ([topicModel, subtopicsCollection, exercisesCollection]) => {
+        const topic = _currentTopicState(topicModel);
+        const subtopics = _topicsState(subtopicsCollection);
+        const exercises = _exercisesState(exercisesCollection);
+
+        store.dispatch('SET_TOPIC', topic);
+        store.dispatch('SET_SUBTOPICS', subtopics);
+        store.dispatch('SET_EXERCISES', exercises);
+        resolve({ topic, subtopics, exercises });
+      },
+      error => reject(error)
+    );
+  });
+}
+
 function showCreateExamPage(store, classId, channelId) {
   store.dispatch('CORE_SET_PAGE_LOADING', true);
   store.dispatch('SET_PAGE_NAME', Constants.PageNames.CREATE_EXAM);
+  store.dispatch('CORE_SET_TITLE', ('Exams'));
```

Comment:
Might be best to start separating out the core titles to an object in your constants file